sw/device/lib/runtime/ibex.h

```diff
@@ -25,20 +25,14 @@
  * Adapted from: The RISC-V Instruction Set Manual, Volume I: Unprivileged ISA
  * V20191213, pp. 61.
  */
-inline uint64_t ibex_mcycle_read(void) {
-  uint32_t cycle_low = 0;
-  uint32_t cycle_high = 0;
-  uint32_t cycle_high_2 = 0;
-  asm volatile(
-      "read%=:"
-      "  csrr %0, mcycleh;"     // Read `mcycleh`.
-      "  csrr %1, mcycle;"      // Read `mcycle`.
-      "  csrr %2, mcycleh;"     // Read `mcycleh` again.
-      "  bne  %0, %2, read%=;"  // Try again if `mcycle` overflowed before
-                                // reading `mcycleh`.
-      : "+r"(cycle_high), "=r"(cycle_low), "+r"(cycle_high_2)
-      :);
-  return (uint64_t)cycle_high << 32 | cycle_low;
+static inline uint64_t ibex_mcycle_read(void) {
```

Comment:
`-O0` or similar might mean the code sequence you want doesn't get produced, so perhaps `ibex_mcycle_read` should be left as-is.